### PR TITLE
Page Cleanup overrides

### PR DESCRIPTION
**Pull request by C-Thomp**

### CODE REVIEW

File: test/pages/CommodityCodePageSpec.scala [hunk 1]
```diff
@@ -28,4 +31,19 @@ class CommodityCodePageSpec extends PageBehaviours {
...
+
+        val ua = emptyUserAnswers
+          .set(CommodityCodePage, "123456")
+          .get
+
+        HasCommodityCodePage.cleanup(Some(false), ua) mustBe
+          Success(emptyUserAnswers)
+      }
+    }
+  }
```
Comment (RawToast) [inline test/pages/CommodityCodePageSpec.scala:48]: I feel like this should be in `HasCommodityCodePageSpec`

Comment (C-Thomp) [reply test/pages/CommodityCodePageSpec.scala]: Agreed

**Review by koladeadewuyi — DISMISSED** [review]

File: test/pages/AreThereRestrictionsOnTheGoodsPageSpec.scala [hunk 2]
```diff
@@ -28,4 +31,18 @@ class AreThereRestrictionsOnTheGoodsPageSpec extends PageBehaviours {
...
+
+        val ua = emptyUserAnswers
+          .set(DescribeTheRestrictionsPage, "restrictions")
+          .get
+
+        AreThereRestrictionsOnTheGoodsPage.cleanup(Some(false), ua) mustBe
+          Success(emptyUserAnswers)
+      }
+    }
+  }
```
Comment (amaalali) [inline test/pages/AreThereRestrictionsOnTheGoodsPageSpec.scala:47]: We want to be testing all  scenarios here, namely what happens when the cleanup is called with a `None`, and `Some(true)`.

File: test/pages/HasCommodityCodePageSpec.scala [hunk 3]
```diff
@@ -28,4 +31,18 @@ class HasCommodityCodePageSpec extends PageBehaviours {
...
+        val emptyUserAnswers = UserAnswers("id", DraftId(1))
+
+        val ua = emptyUserAnswers
+          .set(CommodityCodePage, "123456")
+          .get
+
+        HasCommodityCodePage.cleanup(Some(false), ua) mustBe
+          Success(emptyUserAnswers)
+      }
+    }
```
Comment (amaalali) [inline test/pages/HasCommodityCodePageSpec.scala:46]: We want to be testing all scenarios here, namely what happens when the cleanup is called with a None, and Some(true).

File: test/pages/HasConfidentialInformationPageSpec.scala [hunk 4]
```diff
@@ -28,4 +31,18 @@ class HasConfidentialInformationPageSpec extends PageBehaviours {
...
+      "when HasConfidentialInformationPage is changed to No" in {
+        val emptyUserAnswers = UserAnswers("id", DraftId(1))
+
+        val ua = emptyUserAnswers
+          .set(ConfidentialInformationPage, "secret")
+          .get
+
+        HasConfidentialInformationPage.cleanup(Some(false), ua) mustBe
+          Success(emptyUserAnswers)
+      }
```
Comment (amaalali) [inline test/pages/HasConfidentialInformationPageSpec.scala:45]: We want to be testing all scenarios here, namely what happens when the cleanup is called with a None, and Some(true).

File: test/pages/HaveTheGoodsBeenSubjectToLegalChallengesPageSpec.scala [hunk 5]
```diff
@@ -28,4 +31,18 @@ class HaveTheGoodsBeenSubjectToLegalChallengesPageSpec extends PageBehaviours {
...
+        val emptyUserAnswers = UserAnswers("id", DraftId(1))
+
+        val ua = emptyUserAnswers
+          .set(DescribeTheLegalChallengesPage, "legalities")
+          .get
+
+        HaveTheGoodsBeenSubjectToLegalChallengesPage.cleanup(Some(false), ua) mustBe
+          Success(emptyUserAnswers)
+      }
+    }
```
Comment (amaalali) [inline test/pages/HaveTheGoodsBeenSubjectToLegalChallengesPageSpec.scala:46]: We want to be testing all scenarios here, namely what happens when the cleanup is called with a None, and Some(true).

File: test/pages/HaveYouUsedMethodOneForSimilarGoodsInPastPageSpec.scala [hunk 6]
```diff
@@ -28,4 +31,18 @@ class HaveYouUsedMethodOneForSimilarGoodsInPastPageSpec extends PageBehaviours {
...
+
+        val ua = emptyUserAnswers
+          .set(DescribeTheSimilarGoodsPage, "similarities")
+          .get
+
+        HaveYouUsedMethodOneForSimilarGoodsInPastPage.cleanup(Some(false), ua) mustBe
+          Success(emptyUserAnswers)
+      }
+    }
+  }
```
Comment (amaalali) [inline test/pages/HaveYouUsedMethodOneForSimilarGoodsInPastPageSpec.scala:47]: We want to be testing all scenarios here, namely what happens when the cleanup is called with a None, and Some(true).

File: test/pages/HaveYouUsedMethodOneInPastPageSpec.scala [hunk 7]
```diff
@@ -28,4 +31,18 @@ class HaveYouUsedMethodOneInPastPageSpec extends PageBehaviours {
...
+
+        val ua = emptyUserAnswers
+          .set(DescribeTheIdenticalGoodsPage, "identicalities")
+          .get
+
+        HaveYouUsedMethodOneInPastPage.cleanup(Some(false), ua) mustBe
+          Success(emptyUserAnswers)
+      }
+    }
+  }
```
Comment (amaalali) [inline test/pages/HaveYouUsedMethodOneInPastPageSpec.scala:47]: We want to be testing all scenarios here, namely what happens when the cleanup is called with a None, and Some(true).

File: test/pages/IsSaleBetweenRelatedPartiesPageSpec.scala [hunk 8]
```diff
@@ -28,4 +31,18 @@ class IsSaleBetweenRelatedPartiesPageSpec extends PageBehaviours {
...
+        val emptyUserAnswers = UserAnswers("id", DraftId(1))
+
+        val ua = emptyUserAnswers
+          .set(ExplainHowPartiesAreRelatedPage, "reasons")
+          .get
+
+        IsSaleBetweenRelatedPartiesPage.cleanup(Some(false), ua) mustBe
+          Success(emptyUserAnswers)
+      }
+    }
```
Comment (amaalali) [inline test/pages/IsSaleBetweenRelatedPartiesPageSpec.scala:46]: We want to be testing all scenarios here, namely what happens when the cleanup is called with a None, and Some(true).

File: test/pages/IsTheSaleSubjectToConditionsPageSpec.scala [hunk 9]
```diff
@@ -28,4 +31,18 @@ class IsTheSaleSubjectToConditionsPageSpec extends PageBehaviours {
...
+
+        val emptyUserAnswers = UserAnswers("id", DraftId(1))
+        val ua               = emptyUserAnswers
+          .set(DescribeTheConditionsPage, "conditions")
+          .get
+
+        IsTheSaleSubjectToConditionsPage.cleanup(Some(false), ua) mustBe
+          Success(emptyUserAnswers)
+      }
+    }
```
Comment (amaalali) [inline test/pages/IsTheSaleSubjectToConditionsPageSpec.scala:46]: We want to be testing all scenarios here, namely what happens when the cleanup is called with a None, and Some(true).

**Review by amaalali — CHANGES_REQUESTED** [review]
I think we missing some test cases on these changes

File: test/pages/AreThereRestrictionsOnTheGoodsPageSpec.scala (hunk 2 above)
Comment (C-Thomp) [reply test/pages/AreThereRestrictionsOnTheGoodsPageSpec.scala:47]: Good call, adding now

**Review by amaalali — APPROVED** [review]

**Pull request merged**